Joohwan Ko/partner_selector.py

```diff
@@ -0,0 +1,175 @@
+"""
+Module for selecting partners at Initialization Period
+
+@author: Joohwan Ko
+"""
+import pandas as pd
+import numpy as np
+import seaborn as sns
+import matplotlib.pyplot as plt
+from itertools import combinations
+from tqdm import tqdm
+from statsmodels.distributions.empirical_distribution import ECDF
+
+
+class SelectPartner:
+    """
+    Implementation for section 3.1.1: Partner selection from the paper "Statistical arbitrage with vine copulas(2016,Stübinger)
+    and this handles 4 different ways of selecting partner at Initialization Period
+    1. Traditional Approach
+    2. Extended Approach
+    3. Geometric Approach
+    4. Extremal Approach
+    """
+
+    def __init__(self, rank_df):
+        self.rank_df = rank_df
+        self.rank_corr_df = rank_df.corr()
+
+    def get_partner(self, method='traditional', extended_rho='first'):
+        """
+        Select partners for copula by choosing a method
+        :param method: (str) {'traditional', 'extended', 'geometric', 'extremal'}, default: 'traditional'
+        :param extended_rho: (str) {'first', 'second', 'third'}, default: 'first'
+        :return: (list) Output matrix quadruple Q
+        """
+        Q_matrix = []
```

Comment:
Why are you adding a capital letter? This is confusing. It should be reserved for classes.